src/core/Akka/Actor/FSM.cs

```diff
@@ -178,22 +245,26 @@ public Failure(object cause)
             /// <summary>
             /// TBD
             /// </summary>
-            public object Cause { get; private set; }
+            public object Cause { get; }
         }
 
         /// <summary>
         /// Used in the event of a timeout between transitions
         /// </summary>
-        public class StateTimeout { }
+        public sealed class StateTimeout
+        {
+            private StateTimeout() { }
```

Comment:
This one could break compatibility. Probably I should keep a public constructor with `ObsoletedAttribute`